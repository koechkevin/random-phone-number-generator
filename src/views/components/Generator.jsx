import React, { Component } from 'react';

class Generator extends Component {
  render() {
    return (
      <div className="gen">
        <form onSubmit={(e) => { e.preventDefault(); }}>
          Enter the number of phone numbers to generate
          <br/>
          <input type="number"/>
          <input type="submit" value="Generate"/>
        </form>
      </div>
    );
  }
}

export default Generator;
