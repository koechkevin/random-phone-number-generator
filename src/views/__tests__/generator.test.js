import { mount } from 'enzyme/build';
import React from 'react';
import Generator from '../components/Generator';

describe('test', () => {
  const wrapper = mount(
    <Generator/>,
  );
  it('passes', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('simulates submit', () => {
    const form = wrapper.find('form');
    expect(form.length).toEqual(1);
    form.simulate('submit', {
      target: {
        preventDefault: jest.fn(),
      },
    });
    expect(wrapper.state()).toBe(null);
  });
});
