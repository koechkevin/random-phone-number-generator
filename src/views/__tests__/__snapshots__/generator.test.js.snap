// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test passes 1`] = `ReactWrapper {}`;
